components/list/list-item-generic-layout.js

```diff
@@ -208,17 +208,13 @@ class ListItemGenericLayout extends RtlMixin(LitElement) {
 		if (!cell) return;
 
 		const firstFocusable = getFirstFocusableDescendant(cell);
-		if (!firstFocusable) {
-			const listItem = findComposedAncestor(this, node => node.role === 'rowgroup');
-			if (listItem) {
-				const nextFocusable = previous ? getPreviousFocusable(listItem) : getNextFocusable(listItem);
```

Comment:
This was the culprit - if the adjacent list-item did not have a focusable in the corresponding cell, then it would fall back to this very general method of locating the next/previous focusable.